CONTRIBUTING.md

```diff
@@ -33,16 +37,32 @@ $ git clone git@github.com:<user>/kubebuilder.git $GOPATH/src/sigs.k8s.io/kubebu
 1. Build the project by using the command `make build` 
 1. Run the command `make install` to create a bin with the source code 
 
-## How to test kubebuilder locally
+**NOTE** It will allow you to test the changes made since when you run `kubebuilder` commands, it will use the binary built from your source code. Run `which kubebuilder` to check its path.
+
+## How to verify your local setup
+
+Run the tests by using the command `make go-test`. It will execute unit tests, and then you will be able to check your local configuration.
 
-1. Run the tests by using the command `make test`. It will execute unit tests. 
-1. Run the script `make generate` to update/generate the mock data used in the e2e test in `$GOPATH/src/sigs.k8s.io/kubebuilder/testdata/`
+## What to before to push an pull request  
+
+1. Run the script `make generate` to update/generate the mock data used in the e2e test in `$GOPATH/src/sigs.k8s.io/kubebuilder/testdata/` 
+2. Run `make lint` to check your code.
+
+**NOTE** To run `make lint` is required to install `golangci-lint` locally. More info: https://github.com/golangci/golangci-lint#install
```

Comment:
Instead of all the changes you did here, I would just add this as the third element in the old `How to test kubebuilder locally` and then add the table you created.

Reply:
IMO the other changes are required to clarify the information for new contributors.

Reply:
Usually, contributors do not know what they should do before doing a Pull Request. Also, note that the `make generate` is not a test and is something very specific to the project which requires further explanation. In this way, it cannot be in `How to test kubebuilder locally`.

Reply:
`make generate` actually is a test, the golden one. It is very specific, yes, but it is still a test. By separating it out into two sections seems as if part of those tests do not need to be performed before pull requests.

Reply:
The `make generate` is NOT the test. It will generate the data. (golden.sh)
The  `make test-gen` will call the `golden_test.sh` which is a check/test.

Reply:
True, `make test-gen` is the actual test. I still think adding this in three sections just makes the process harder to udnerstand to new contributors. Test setup (like `make generate`) and tests themselves (like `make test-gen`) are better in a single section IMO.

Reply:
It is not in 3 sections at all.  The `make generate` is just in the section `## What to before to push an pull request`.

Reply:
Actually, they are 4 sections:
```
## How to build kubebuilder locally
## How to verify your local setup
## What to before to push an pull request
## How to test kubebuilder locally
```

Reply:
But are NOT the sections related to `make generate`. 
Each section is an diff topic and has a different purpose.

Reply:
PS.: I understand that you are against the sections. However, it is the previous standard of this doc as well as the other ones in the repo. I do not think that we should have 1 section with all.

Reply:
There were two sections, four seems to be too much. They are small and the last 3 are rlated to the tests.

Reply:
Done changes to address your concern.  Let me know if you are ok with now.

Reply:
Marking it as closed.